addons/l10n_vn/models/res_bank.py

```diff
@@ -0,0 +1,65 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ResPartnerBank(models.Model):
+    _inherit = 'res.partner.bank'
+
+    proxy_type = fields.Selection(selection_add=[('merchant_id', 'Merchant ID'), ('atm_card', 'ATM Card Number'),
+                                                 ('bank_acc', 'Bank Account')],
+                                  ondelete={'merchant_id': 'set default', 'atm_card': 'set default', 'bank_acc': 'set default'})
+
+    @api.constrains('proxy_type', 'proxy_value', 'partner_id')
```

Comment:
```suggestion
    @api.constrains('proxy_type')
```
These are not used in the check, so no need to trigger it when these changes.